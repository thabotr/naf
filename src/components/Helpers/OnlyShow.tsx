export function OnlyShow({
  If,
  children,
}: {
  If?: boolean;
  children: React.ReactNode;
}) {
  return If ? <>{children}</> : null;
}
