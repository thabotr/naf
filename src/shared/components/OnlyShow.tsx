import React from 'react';

export default function OnlyShow({
  If,
  children,
}: {
  If: boolean;
  children: React.ReactNode;
}) {
  return If ? <>{children}</> : null;
}
