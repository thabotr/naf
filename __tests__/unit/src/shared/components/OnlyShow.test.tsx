import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import OnlyShow from '../../../../../src/shared/components/OnlyShow';

describe('OnlyShow', () => {
  it('correctly renders its child component when the If prop is truthy', () => {
    const child = <Text>Im am the child component</Text>;
    const tree = renderer
      .create(<OnlyShow If={true}>{child}</OnlyShow>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('does not render its child component when the If prop is falsy', () => {
    const child = <Text>Testing Text</Text>;
    const tree = renderer
      .create(<OnlyShow If={false}>{child}</OnlyShow>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
