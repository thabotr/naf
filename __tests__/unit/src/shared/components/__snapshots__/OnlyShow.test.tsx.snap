// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlyShow correctly renders its child component when the If prop is truthy 1`] = `
<Text>
  Im am the child component
</Text>
`;

exports[`OnlyShow does not render its child component when the If prop is falsy 1`] = `null`;
